spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/ProtobufUtils.java

```diff
@@ -581,6 +600,15 @@ static StructField[] createMapStructFields(MapType mapType) {
   }
 
   private static DescriptorProtos.FieldDescriptorProto.Type toProtoFieldType(DataType sparkType) {
+    Optional<DescriptorProtos.FieldDescriptorProto.Type> protoFieldType =
```

Comment:
Consider refactoring and removing the code duplication

Reply:
Refactored the code